components/eamxx/src/share/io/scream_io_file_specs.hpp

```diff
@@ -86,7 +86,10 @@ struct IOFileSpecs {
   // If positive, flush the output file every these many snapshots
   int flush_frequency = std::numeric_limits<int>::max();
 
-  // bool file_is_full () const { return num_snapshots_in_file>=max_snapshots_in_file; }
+  bool file_is_full () const { 
+    return storage.num_snapshots_in_file>=storage.max_snapshots_in_file; 
```

Comment:
This only works if the storage type is based on max number of snapshots. It will not work in case one chooses "one_month" or "one_year" (the former being quite appealing for single month data).

Reply:
The fact that this only works for max-snaps based storage may very well be the case why it was commented out.

Reply:
This is an excellent point.  Thank you for pointing this out.

Reply:
If the simulation writes a restart file and the rpointers are all consistent, and then the simulation fails, will a file that is still open (for a year, for example) still have the risk of containing 0s instead of flushed data in a time period that is prior to the valid restart? I.e., could we still end up with 0 data?

Perhaps this is already done, but it seems to me the only way to guarantee things is by flushing _all_ files prior to writing the rpointer file.

Reply:
@ambrad we also have a separate todo item: every time we write a rhist file, also flush the corresponding output file.

Reply:
Is it possible for a file to be open during restart and not having an associated rhist file?

Reply:
@ambrad was this just a question to ensure we're not forgetting anything, or do you have a scenario where you think that would/should be the case (so flushing at rhist write would not be enough)?

Reply:
Could we have a `flush_all` type call that is initiated whenever a restart is written?  Wouldn't that remedy the concern of having all 0's if a fail occurs after a restart is written?

Reply:
@bartgol, right, to ensure we're not forgetting anything. But I agree with Aaron that if the AD has a list of open write-mode files, why not just iterate through the list and flush them all before writing the rpointer file? You'd then skip more file-specific flushing. I don't think there can be an open write-mode file that _shouldn't_ be flushed at a restart write.

Reply:
I don't think the AD has a handle to all files _directly_. But each output manger has a handle to its output and rhist files, so when the rhist file is written, it can flush the output one too.

And as Naser said, in EAMxx (for some technical reasons) we _always_ write a rhist file, even if there is no "restart data" (e.g., for INSTANT output) and even if we just wrote in the output file. So flushing the .h file  when the corresponding  .rhist is flushed, should cover every output file.